qcodes/instrument/meta/meta_instrument.py

```diff
@@ -0,0 +1,185 @@
+from typing import List, Dict, Union, Any, Optional
+
+from functools import partial
+
+from qcodes.instrument.meta.meta_parameter import MetaGroup, MetaParameter
+from qcodes.instrument.parameter import Parameter
+from qcodes.instrument.base import InstrumentBase
+from qcodes.station import Station
+
+import logging
+
+_log = logging.getLogger(__name__)
+
+
+class MetaInstrument(InstrumentBase):
+    """MetaInstrument class for creating a meta instrument that aliases one or more 
+    parameter endpoints from real instruments.
+
+    Example usage in instrument YAML:
+
+    field:
+        type: qcodes.instrument.meta.MetaInstrument
+        init:
+        aliases:
+            X:
+            - field_X.field
+            ramp_rate:
+            - field_X.ramp_rate
+        set_initial_values_on_load: true
+        initial_values:
+            ramp_rate: 0.02
+        setters:
+            X:
+            method: field_X.set_field
+            block: false
+        units:
+            X: T
+            ramp_rate: T/min
+
+    Args:
+        name: Instrument name
+        station: Real instrument station to connect the endpoints to.
+        aliases: Aliases and the endpoints they connect to.
```

Comment:
For example this should say something like. A mapping from name of an alias parameter to the sequence of endpoints that it connects to.